packages/mui-base/src/SliderUnstyled/useSlider.ts

```diff
@@ -625,6 +612,38 @@ export default function useSlider(props: UseSliderProps) {
     };
   };
 
+  const createHandleMouseOver =
+    (otherHandlers: Record<string, React.EventHandler<any>>) =>
+    (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+      const index = Number(event.currentTarget.getAttribute('data-index'));
+      setOpen(index);
+
+      otherHandlers.onMouseOver?.(event);
```

Comment:
I think that we should call the external handlers before our internal logic. This would make it more similar to the native web controls. This way we could also allow developers to opt out of running the internal handler (by using preventDefault), as I suggested above.

Reply:
Makes sense, let me refactor

Reply:
Please check https://github.com/mui-org/material-ui/pull/30094/commits/9f835697cc3c582e5f821873ba725ec9a08adbef if that's what you had in mind. It makes sense to me.

Reply:
Yep, that's what I meant :)